metals/src/main/scala/scala/meta/internal/troubleshoot/ProblemResolver.scala

```diff
@@ -252,6 +261,32 @@ class ProblemResolver(
           "-Xplugin:semanticdb -targetroot:javac-classes-directory"
         )
       )
-    else None
+    else isWrongJavaRelease(javaTarget, scalaTarget)
+  }
+
+  private def isWrongJavaRelease(
+      javaTarget: JavaTarget,
+      scalaTarget: Option[ScalaTarget]
+  ): Option[JavaProblem] = {
+    def buildJavaVersion =
+      for {
+        target <- scalaTarget
+        javaHome <- target.jvmHome
+        path = javaHome.toAbsolutePath
+        version <- JdkVersion.getJavaVersionFromJavaHome(path)
+      } yield version
+
+    val releaseVersion = javaTarget.releaseVersion.flatMap(JdkVersion.parse)
+    releaseVersion.zip(buildJavaVersion) match {
+      case Some((releaseVersion, jvmHomeVersion))
+          if jvmHomeVersion.major < releaseVersion.major || jvmHomeVersion.minor < releaseVersion.minor =>
```

Comment:
I'm not convicted about `|| jvmHomeVersion.minor < releaseVersion.minor` Shouldn't minor version be compatible?

Reply:
minor in this case is if we get `1.8` instead of `8`. This might not be needed at all really, usually we only set major.

But maybe I should switch the condition a bit.

Reply:
@tgodzik I think we can align versions to a single format in `JdkVersion`.

Reply:
I switched to just using major. We don't actually need the minor, so it should be ok now.

Reply:
@tgodzik thx!